duckduckgo/macos-browser
# Smarter Encryption library integration

Task/Issue URL: https://app.asana.com/0/1193060753475688/1178859670483593
CC: @tomasstrba
Reference PR: https://github.com/duckduckgo/iOS/pull/749

**Description**:

This PR adds support for Smarter Encryption via an embedded copy of the bloom filter binary file. Specifically:

1. The Bloom Filter submodule has been added
1. The HTTPS upgrade Core Data model is added
1. Support for loading the Bloom Filter data and using it to determine whether a URL should be upgraded

Support for fetching the latest bloom filter file will come in a future project.

**Steps to test this PR**:

1. Navigate to http://facebook.com and ensure this is upgraded
1. Navigate to some long URL that doesn't exist and is thus definitely not in our list e.g http://thisUrlDoesNotExistAndIsNotInOurSmarterEncryptionList.com and ensure this is not upgraded
